Disable install button for all add-ons on Fenix

bobsilverberg opened the pull request:
Fixes #9867 
Fixes #9864

eviljeff commented on src/core/utils/compatibility.js:
```diff
@@ -166,6 +166,13 @@ export function isCompatibleWithUserAgent({
 
+  // Fenix does not support add-ons (yet?).
+  // See: https://github.com/mozilla-mobile/fenix/issues/1134
+  // See also: https://github.com/mozilla/addons-frontend/issues/7963
```
https://github.com/mozilla/addons-frontend/issues/9864 is probably a better issue to mention?

bobsilverberg replied:
Yeah, that was a copy and paste of the code that was there before the patch that removed it. I'll update it.

eviljeff commented on src/core/utils/compatibility.js:
```diff
@@ -166,6 +166,13 @@ export function isCompatibleWithUserAgent({
     return { compatible: false, reason: INCOMPATIBLE_NOT_FIREFOX };
   }
 
+  // Fenix does not support add-ons (yet?).
```
mmmmmmmmmm, that's not really true.

bobsilverberg replied:
Yeah, we should change it to say add-ons are not installable from AMO on Fenix yet.

eviljeff commented on src/core/utils/compatibility.js:
```diff
@@ -166,6 +166,13 @@ export function isCompatibleWithUserAgent({
+  // See: https://github.com/mozilla-mobile/fenix/issues/1134
+  // See also: https://github.com/mozilla/addons-frontend/issues/7963
+  if (isFirefoxForAndroid(userAgentInfo)) {
+    return { compatible: false, reason: INCOMPATIBLE_ANDROID_UNSUPPORTED };
```
so this replaces the more nuanced logic below that checks the OS and the compatibility by just making all add-ons incompatible with Fennec on Android regardless of the compatibility status?

bobsilverberg replied:
That is correct. For now, we want all add-ons to appear incompatible if someone is running Firefox for Android. There's an issue to change this, when they do become installable, at which point we can being some of the logic back.

willdurand replied:
couldn't we use a feature flag instead then?

bobsilverberg replied:
Yeah, I can probably get that done in time for the tag. This was an attempt at a quick fix for a bug that is currently present in production. I didn't know if we were going to have to hotfix it or not, so I just wanted a patch that fixes it.

Now that I know we can wait until Thursday for this fix, we can probably address #9864 in this patch as well.

bobsilverberg commented:
@eviljeff I have addressed all of the comments, and also added a flag to address #9864. I also did a bit of extra cleanup in the test helpers and tests as we no longer differentiate between Fennec and Fenix, so we shouldn't have anything directly referencing Fenix.

eviljeff commented on tests/unit/core/utils/test_compatibility.js:
```diff
@@ -1115,12 +1127,6 @@ describe(__filename, () => {
   });
 
   describe('isFirefoxForIOS', () => {
-    it('returns false for Firefox Fenix', () => {
```
😕 FirefoxForIOS was testing for Fenix but now isn't?

bobsilverberg replied:
As discussed below, any tests that existed for Fenix have now been combined into tests for Firefox for Android.

eviljeff reviewed (commented):
I don't understand the changes in test_compatablity and /helpers.js - fenix is now fennec??  Where did that happen in the changes in the live code?

bobsilverberg commented:
> I don't understand the changes in test_compatablity and /helpers.js - fenix is now fennec?? Where did that happen in the changes in the live code?

Both Fenix and Fennec are now Firefox for Android. A lot of the changes regarding that were actually made in earlier patches, but I neglected to fix up the tests. Because we are assuming that anyone on an Android browser is using Fenix, there were tests added specifically for Fenix, but that doesn't really make sense, as nothing in the code base differentiates between Fennec and Fenix anymore.

This is why I combined both sets of user agents into `android` in `tests/unit/helpers.js`, and also why I removed any test cases that were specifically for Fenix. We don't need any of those anymore. Instead we just need a test case for Firefox for Android, and we already had those in place.

bobsilverberg commented on src/core/utils/compatibility.js:
```diff
@@ -166,6 +169,18 @@ export function isCompatibleWithUserAgent({
+      return { compatible: false, reason: INCOMPATIBLE_ANDROID_UNSUPPORTED };
+    }
+    // We need to check that the add-on is installable.
+    if (!isAndroidInstallable({ addon })) {
```
Hmm, I suppose we should still be checking that. It's annoying that there are two sets of criteria to specify that an extension can be installed on Android (`currentVersion.compatibility[CLIENT_APP_ANDROID]` and `promoted`), but I guess both are still valid. I'll add that back in.

bobsilverberg replied:
I added this back in. I accidentally did an `amend` commit so it doesn't show up in a separate commit.
